### PR TITLE
[PLAT-8394] Fix crash in minified Android builds

### PR DESCRIPTION
**Pull request by nickdowell**
## Goal

Fix a crash in Bugsnag when used in Android apps that are built [for distribution](https://docs.unrealengine.com/5.0/en-US/signing-android-projects-for-release-on-the-google-play-store-with-unreal-engine/), with the following log messages:

```
W/System.err(26556): java.lang.ClassNotFoundException: Didn't find class "com.bugsnag.android.App" on path: ...
W/System.err(26556): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:207)
```

## Changeset

Adds a ProGuard rule (via UPL) to prevent `com.bugsnag.**` Java classes being obfuscated.

Makes the test fixture's `gameActivityClassAdditions` methods public so that they are not hidden by ProGuard.

## Testing

Updates build files to produce Distribution (minified) and signed builds for example app & test fixture. This reproduces the crash and causes E2E testing to fail - e.g. https://buildkite.com/bugsnag/bugsnag-unreal/builds/1014

E2E tests pass with the fix in place.

### CODE REVIEW

File: features/fixtures/generic/Plugins/Tweaks/Source/Tweaks/Tweaks_UPL.xml [hunk 1]
```diff
@@ -21,14 +21,14 @@
 
     <gameActivityClassAdditions>
         <insert>
-            private void clearBugsnagCache() {
+            public void clearBugsnagCache() {
```
Comment (kattrali) [inline features/fixtures/generic/Plugins/Tweaks/Source/Tweaks/Tweaks_UPL.xml:24]: What is this change for?

**Review by kattrali — APPROVED** [review]
looks fine, saw maybe one spurious change?

File: features/fixtures/generic/Plugins/Tweaks/Source/Tweaks/Tweaks_UPL.xml (hunk 1 above)
Comment (nickdowell) [reply features/fixtures/generic/Plugins/Tweaks/Source/Tweaks/Tweaks_UPL.xml:24]: ProGuard seems to hide private methods, making them unaccessible from JNI.

Comment (kattrali) [reply features/fixtures/generic/Plugins/Tweaks/Source/Tweaks/Tweaks_UPL.xml:24]: Hmm. So this is simpler than adding it to the app's proguard rules. Ok :+1:

Comment (nickdowell) [reply features/fixtures/generic/Plugins/Tweaks/Source/Tweaks/Tweaks_UPL.xml:24]: Yes it seemed simpler to make these public than adding more proguard rules 🙃

**Pull request merged**